src/WebJobs.Script.SiteExtension/New-PrivateSiteExtension.ps1

```diff
@@ -0,0 +1,130 @@
+#>
+
+param (
+    [string] $InputPath = $null,
+    [string] $OutputPath = $null,
+    [ValidateSet('x64', '64bit', 'x86', '32bit')][string] $Bitness = '32bit',
+    [switch] $Zip,
+    [switch] $Force
+)
+
+$normalizeBitness = @{
+    'x64' = '64bit'
+    '64bit' = '64bit'
+    'x86' = '64bit'
+    '32bit' = '32bit'
+}
+
+$Bitness = $normalizeBitness[$Bitness]
+
+if (-not $InputPath)
+{
+    $InputPath = (Get-ChildItem -Path . -Filter "extension.xml" -Recurse).Directory.FullName
+}
+
+if (Test-Path (Join-Path $InputPath "WebJobs.Script.SiteExtension.csproj"))
+{
+    Write-Error "This script should not be ran in the WebJobs.Script.SiteExtension project folder. Run this script in the root of the published site extension folder."
+    exit 1
+}
+
+if (-not (Join-Path $InputPath "extension.xml" | Test-Path))
+{
+    Write-Error "Unable to find published site extension."
+    exit 1
+}
+
+if (-not $OutputPath)
+{
+    $runtime = $Bitness -eq '32bit' ? 'win-x86' : 'win-x64'
+    $leaf = (Split-Path $InputPath -Leaf)
+    $split = $leaf.IndexOf('.')
+    $OutputPath = "$($leaf.Substring(0, $split)).Private.$($leaf.Substring($split + 1)).$runtime"
+}
+
+function New-TemporaryDirectory {
+    $tmp = [System.IO.Path]::GetTempPath()
+    $name = (New-Guid).ToString("N")
+    return New-Item -ItemType Directory -Path (Join-Path $tmp $name)
+}
+
+function Write-Zip ($outputPath)
+{
+    if (Test-Path $outputPath) {
+        if ($Force) {
+            Remove-Item -Path $outputPath -Recurse -Force
+        } else {
+            Write-Error "OutputPath already exists. Use -Force to overwrite."
+            exit 1
+        }
+    }
+
+    $tempDir = New-TemporaryDirectory
+    Write-Folder $tempDir
+
+    Compress-Archive -Path "$tempDir/*" -DestinationPath $outputPath
+    Remove-Item -Path $tempDir -Recurse -Force
+}
+
+function Write-Folder ($outputPath)
+{
+    if (Test-Path "$outputPath/*") {
+        if ($Force) {
+            Remove-Item -Path $outputPath -Recurse -Force
+        } else {
+            Write-Error "OutputPath already exists. Use -Force to overwrite."
+            exit 1
+        }
+    }
+
+    $inputPath = Get-ChildItem -Path $InputPath -Attributes Directory
+    $outputPath = Join-Path $outputPath "SiteExtensions" "Functions"
+    New-Item -ItemType Directory -Path $outputPath | Out-Null
+
+    Copy-Item "$inputPath/applicationHost.xdt" -Destination $outputPath | Out-Null
+
+    $filesDest = (Join-Path $outputPath "$Bitness")
+    Copy-Item "$inputPath/$Bitness/" -Destination "$filesDest/" -Container -Recurse | Out-Null
+
+    $workerDest = (Join-Path $filesDest "workers")
+    Copy-Item "$inputPath/workers/" -Destination "$workerDest/" -Container -Recurse | Out-Null
+}
+
+if ($Zip) {
+    if (-not $OutputPath.EndsWith(".zip")) {
+        $OutputPath = "$OutputPath.zip"
+    }
+
+    Write-Zip $OutputPath
+} else {
+    Write-Folder $OutputPath
+}
+
+Write-Host "Published private site extension to $OutputPath"
```

Comment:
Have you tested out this site extension in the azure portal to confirm this work?

Reply:
No I have not - the structure of it should be identical to before, just the way it is built is different.

Reply:
I gave this a try and my func app is in error - not clear yet if thats just the host version that was built in the PR, or the private site extension. Will update my findings here soon.

Reply:
![Screenshot 2025-01-22 at 5 36 29 PM](https://github.com/user-attachments/assets/b65f03e9-91a1-4a33-9e8d-941750aea20c)

Actually, I do see 64bit on the platform here, maybe private site extensions do work with this and the documentation for that is just outdated? Only way to double check is to test both 32 and 64 bit pse deploement

Reply:
Okay, looks like it's working. Bitness defaults to 64 and I had to change it 32 to match the pse I built/published.

```
{
    "id": "<MYAPP>",
    "state": "Running",
    "version": "4.1037.0.25058",
    "versionDetails": "4.1037.0-ci.25058.2+9adf37ff9a0fcaf91c3338b6d10a52b242be68ac",
    "platformVersion": "103.0.7.240",
    "instanceId": "3b13ece7e0f3c09a90939daeaed59824c94907a7f26aeebdcab6fa7b282a2eca",
    "computerName": "dw1sdwk0000HV",
    "processUptime": 181107,
    "functionAppContentEditingState": "Unknown",
    "extensionBundle": {
        "id": "Microsoft.Azure.Functions.ExtensionBundle",
        "version": "4.19.0"
    }
}
```

Reply:
And 64bit:

```
{
    "id": "<MYAPP>64bit",
    "state": "Running",
    "version": "4.1037.0.25058",
    "versionDetails": "4.1037.0-ci.25058.2+9adf37ff9a0fcaf91c3338b6d10a52b242be68ac",
    "platformVersion": "103.0.7.240",
    "instanceId": "3b13ece7e0f3c09a90939daeaed59824c94907a7f26aeebdcab6fa7b282a2eca",
    "computerName": "dw1sdwk0000HV",
    "processUptime": 63425,
    "functionAppContentEditingState": "Unknown",
    "extensionBundle": {
        "id": "Microsoft.Azure.Functions.ExtensionBundle",
        "version": "4.19.0"
    }
}
```